private/tools/java/rules/jvm/external/maven/Outdated.java

```diff
@@ -0,0 +1,133 @@
+package rules.jvm.external.maven;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.maven.artifact.versioning.ComparableVersion;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class Outdated {
+  public static String getReleaseVersion(String repository, String groupId, String artifactId) {
+    String releaseVersion = null;
+
+    String url =
+        String.format("%s/%s/%s/maven-metadata.xml",
+            repository,
+            groupId.replaceAll("\\.", "/"),
+            artifactId);
+
+    DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+    DocumentBuilder documentBuilder;
+    try {
+      documentBuilder = documentBuilderFactory.newDocumentBuilder();
+    } catch (ParserConfigurationException e) {
+      verboseLog(String.format("Caught exception %s", e));
+      return null;
+    }
+    Document document;
+    try {
+      document = documentBuilder.parse(new URL(url).openStream());
+    } catch (IOException | SAXException e) {
+      verboseLog(String.format("Caught exception for %s: %s", url, e));
+      return null;
+    }
+
+    // example maven-metadata.xml
+    //<metadata>
+    //  <versioning>
+    //    <latest>1.14.0-SNAPSHOT</latest>
+    //    <release>1.13.0</release>
+    //  </versioning>
+    //</metadata>
+    Element metadataElement = document.getDocumentElement();
+    Element versioningElement = getFirstChildElement(metadataElement, "versioning");
+    if (versioningElement != null) {
+      releaseVersion = versioningElement.getElementsByTagName("release").item(0).getTextContent();
```

Comment:
Note: Another possible feature is allowing people to look for updates from `latest` instead of `release`.